src/clr-angular/data/tree-view/tree-node-link.ts

```diff
@@ -1,16 +1,18 @@
 /*
- * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Directive, ElementRef } from '@angular/core';
+import { Directive, ElementRef, Optional } from '@angular/core';
+import { RouterLinkActive } from '@angular/router';
```

Comment:
I've resisted this before, because it now puts a dependency on the router. Why don't we just have the apps put `aria-selected` themselves so we aren't tying to the router? Some folks use a different router.

Reply:
That's why I used `@Optional()` for the `RouterLinkActive` injection. If it's present, we will manage it. If it's not present, we won't be involved. Also, if the user is using `RouterLinkActive`, it's better for them not to be involved in setting the value of `aria-selected` as it should be always in sync with `RouterLinkActive`'s current state. If they are not relying on `RouterLinkActive`, by all means, they can manage the `aria-selected` on their side. And this behavior is reflected in the website example.

Reply:
Using `@Optional` only works for injection, not for importing. That is why this will fail if you use Tree View without first installing `@angular/router`.

Reply:
oh shoot you are absolutely right! It will make even simple tree nodes dependent on angular router module. I will fix it. Thank you!

Reply:
If we can bump to TS 3.8, it might be possible to use `import()` https://github.com/tc39/proposal-dynamic-import https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#-es2020-for-target-and-module

Reply:
So Angular 10 requires TS 3.8, so we could use that but then we can't backport it.

Reply:
Yeah I think it's too early to adopt dynamic imports. I will just offer doc guidelines as a solution for now.